src/main/java/com/amazonaws/services/kinesis/leases/impl/LeaseCoordinator.java

```diff
@@ -89,32 +90,36 @@
      * Constructor.
      *
      * @param leaseManager LeaseManager instance to use
+     * @param leaseSelector LeaseSelector instance to use
      * @param workerIdentifier Identifies the worker (e.g. useful to track lease ownership)
      * @param leaseDurationMillis Duration of a lease
      * @param epsilonMillis Allow for some variance when calculating lease expirations
      */
     public LeaseCoordinator(ILeaseManager<T> leaseManager,
+            LeaseSelector<T> leaseSelector,
```

Comment:
Adding this parameter will break custom implementations of the LeaseCoordinator.  We need to determine a way to add the parameter without breaking existing implementations.